hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHANodeDetails.java

```diff
@@ -158,7 +166,28 @@ public static SCMHANodeDetails loadSCMHAConfig(OzoneConfiguration conf)
         ScmConfigKeys.OZONE_SCM_DEFAULT_SERVICE_ID);
 
     LOG.info("ServiceID for StorageContainerManager is {}", localScmServiceId);
-
+    if(!storageConfig.isPresent()){
+      storageConfig = Optional.of(new SCMStorageConfig(conf));
+    }
+    Storage.StorageState state = storageConfig.get().getState();
+    boolean scmHAEnableDefault = state == Storage.StorageState.INITIALIZED
+        ? storageConfig.get().isSCMHAEnabled()
+        : SCMHAUtils.isSCMHAEnabled(conf);
+    boolean scmHAEnabled = SCMHAUtils.isSCMHAEnabled(conf);
+    if (Storage.StorageState.INITIALIZED == state
+        && Strings.isNotEmpty(conf.get(ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY))
+        && scmHAEnabled != scmHAEnableDefault) {
+      throw new ConfigurationException(String.format("Invalid Config %s " +
+          "Provided ConfigValue: %s, Expected Config Value: %s",
+          ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY, scmHAEnabled, scmHAEnableDefault));
+    } else if (Storage.StorageState.INITIALIZED == state
+        && scmHAEnabled != scmHAEnableDefault) {
+      LOG.warn("Invalid Config {}, Expected Config Value: {}, Default Config " +
+              "Value: {}", ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY,
+          scmHAEnableDefault, scmHAEnabled);
+    }
```

Comment:
I feel this if-else block can be simplified as follows for readability
```suggestion
String scmHAEnableConfig = conf.get(ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY);
if (state.equals(Storage.StorageState.INITIALIZED) &&
scmHAEnabled != scmHAEnableDefault)){
  if (Strings.isNotEmpty(scmHAEnableConfig)) {
    throw new ConfigurationException(String.format("Invalid Config %s " +
            "Provided ConfigValue: %s, Expected Config Value: %s",
        ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY, scmHAEnabled,
        scmHAEnableDefault));
  } else {
    LOG.warn(
        "Invalid Config {}, Expected Config Value: {}, Default Config " +
            "Value: {}", ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY,
        scmHAEnableDefault, scmHAEnabled);
  }
}
```